src/idns/commands/query.rs

```diff
@@ -0,0 +1,437 @@
+pub struct Query {
+    /// The name of the resource records to look up
+    #[arg(value_name="QUERY_NAME")]
+    qname: Name<Vec<u8>>,
+
+    /// The record type to look up
+    #[arg(value_name="QUERY_TYPE", default_value = "A")]
+    qtype: Rtype,
+
+    /// The server to send the query to. System servers used if missing
+    #[arg(short, long, value_name="ADDR_OR_HOST")]
+    server: Option<ServerName>,
+
+    /// The port of the server to send query to.
+    #[arg(short = 'p', long = "port", requires = "server")]
+    port: Option<u16>,
+
+    /// Use only IPv4 for communication.
+    #[arg(short = '4', long, conflicts_with = "ipv6")]
+    ipv4: bool,
+
+    /// Use only IPv6 for communication.
+    #[arg(short = '6', long, conflicts_with = "ipv4")]
+    ipv6: bool,
+
+    /// Use only TCP.
+    #[arg(short, long)]
+    tcp: bool,
+
+    /// Use the given message ID. Random if missing.
+    #[arg(long)]
+    id: Option<u16>,
+
+    /// Unset the RD flag in the request.
+    #[arg(long)]
+    no_rd: bool,
+
+    /// Disable all sanity checks.
+    #[arg(long, short = 'f')]
+    force: bool,
+
+    /// Set the timeout for a query.
+    #[arg(long, value_name="SECONDS")]
+    timeout: Option<f32>,
+
+    /// Set the number of retries over UDP.
+    #[arg(long)]
+    retries: Option<u8>,
+
+    /// Set the advertised UDP payload size.
+    #[arg(long)]
+    udp_payload_size: Option<u16>,
+
+    /// Verify the answer against an authoritative server.
+    #[arg(long)]
+    verify: bool,
+
+    /// Select the output format.
+    #[arg(long = "format", default_value = "dig")]
+    output_format: OutputFormat,
+}
+
+/// # Executing the command
+///
+impl Query {
+    pub fn execute(self) -> Result<(), Error> {
+        if !self.force {
+            if self.qtype == Rtype::AXFR || self.qtype == Rtype::IXFR {
+                return Err(
+                    "Please use the 'xfr' command for zone transfer.\n\
+                     (Use --force to query anyway.)".into()
+                );
+            }
+        }
+
+        tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap()
+            .block_on(self.async_execute())
+    }
+
+    pub async fn async_execute(self) -> Result<(), Error> {
+        let servers = match self.server {
+            Some(ServerName::Name(ref host)) => self.host_server(host).await?,
+            Some(ServerName::Addr(addr)) => self.addr_server(addr),
+            None => self.default_server(),
+        };
+
+        let request = self.create_request();
+        let answer = self.send_and_receive(servers, request).await?;
+        self.output_format.print(answer.for_slice_ref())?;
+        if self.verify {
+            let auth_answer = self.auth_answer().await?;
+            if Self::eq_answer(
+                answer.for_slice_ref(), auth_answer.for_slice_ref()
+            ) {
+                println!("\n;; Authoritative answer matches.");
```

Comment:
Matches what?

Reply:
The response that was just printed before?

Reply:
Okay, I get it. Perhaps a little more text such as "Authoritative answer from server <IP> matches the answer from server <IP>" might be an improvement, and it might help if it were not lost in the sea of output but instead shown at the end?

Reply:
I think this is a little better now?